src/main/java/com/github/soerxpso/xpspawners/manager/SpawnerManager.java

```diff
@@ -0,0 +1,66 @@
+package com.github.soerxpso.xpspawners.manager;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.block.Block;
+import org.bukkit.inventory.ItemStack;
+
+import com.github.soerxpso.xpspawners.Spawner;
+
+public class SpawnerManager {
+	private HashMap<Long, ArrayList<Spawner>> spawners = 
+			new HashMap<Long, ArrayList<Spawner>>();
+	
+	public ItemStack convertSpawnerToItem(Block b) {
+		return new ItemStack(Material.MOB_SPAWNER);
+	}
+	
+	public Spawner nearestSpawner(Location loc) {
+		Spawner closest = null;
+		for(int xShift = -1; xShift <= 1; xShift++) {
+			for(int zShift = -1; zShift <= 1; zShift++) {
+				long chunkID = chunkID(loc.add(xShift, 0, zShift));
```

Comment:
So ideally, you would use a quadtree or similar -- see https://github.com/Civcraft/CivModCore/blob/master/src/main/java/vg/civcraft/mc/civmodcore/locations/SparseQuadTree.java

The chunk map approach is probably fine for something as sparse as this though.